lib/shopify_api/resources/fulfillment.rb

```diff
@@ -9,5 +9,39 @@ def order_id
     def cancel; load_attributes_from_response(post(:cancel, {}, only_id)); end
     def complete; load_attributes_from_response(post(:complete, {}, only_id)); end
     def open; load_attributes_from_response(post(:open, {}, only_id)); end
+
+    def order_id=(order_id)
+      prefix_options[:order_id] = order_id
+    end
+
+    def load(attributes, remove_root = false, persisted = false)
+      order_id = attributes['order_id']
+      prefix_options[:order_id] = order_id if order_id
+      super(attributes, remove_root, persisted)
+    end
+
+    def save
+      if prefix_options[:order_id].present?
+        super
+      else
+        line_items = attributes['line_items_by_fulfillment_order'] || attributes[:line_items_by_fulfillment_order]
+        if line_items.blank?
+          raise ShopifyAPI::ValidationException,
+                "either 'line_items_by_fulfillment_order' or prefix_options[:order_id] is required"
+        end
+
+        fulfillmentV2 = FulfillmentV2.new(attributes)
+        result = fulfillmentV2.save
```

Comment:
Is this a way of migrating fulfillments? Why are you doing this?

Reply:
We have this situation because the backend has two REST paths for fulfillments, the existing pre-FulfillmentOrder one at `/order/#/fulfillments` and the new FulfillmentOrder based one at `/fulfillments`. To make them interoperable we use the `Fulfillment` resource which routes to the `FulfillmentV2` resource when we need to use the `/fulfillments` path for an operation. Whenever reading a response, we always convert FulfillmentV2 back to Fulfillment.